Rubberduck.Parsing/Symbols/ClassModuleDeclaration.cs

```diff
@@ -210,6 +210,22 @@ private bool HasPredeclaredIdToCache()
 
         public IEnumerable<Declaration> Subtypes => _subtypes.Keys;
 
+        public bool IsInterface => _subtypes.Count > 0;
+
+        public bool IsUserInterface => Subtypes.Any(s => s.IsUserDefined);
+
+        public IEnumerable<ClassModuleDeclaration> ImplementedInterfaces =>
+            _supertypes.Cast<ClassModuleDeclaration>().Where(type => type.Subtypes.Contains(this));
+
+
+        private readonly List<Declaration> _members = new List<Declaration>();
+        public IReadOnlyList<Declaration> Members => _members;
+
```

Comment:
I think this would also make sense on a `ProceduralModuleDeclaration`. Maybe we should extract an abstract `ModuleDeclaration` all module declarations derive from.

Reply:
I think that's a good idea and it would be a really easy change. It would likely provide opportunities for some other code simplifications similar to this PR in the `DeclarationFinder` (and elsewhere).  

I'll implement that when I fix the tests - might as well do it with the hairy merge...